man/adoc/bpftrace.adoc

```diff
@@ -932,6 +933,19 @@ bpftrace knows two types of variables, 'scratch' and 'map'.
 'scratch' variables are kept on the BPF stack and only exists during the execution of the action block and cannot be accessed outside of the program.
 Scratch variable names always start with a `$`, e.g. `$myvar`.
 
+'scratch' variables can also declared before or during initialization with `let` e.g.
+```
+let $x;
```

Comment:
Does it matter? You still cannot use the variable before assigning to it.

Reply:
It does matter (I also added documentation). Think of this program, which I added a runtime test for:
`BEGIN { let $a; if (false) { $a = 1; } @b = $a; exit();}` - semantic analyzer won't catch this (and it probably never could, except in silly cases like this, because it would need to evaluate the conditional).

Reply:
Shouldn't this be a semantic analysis error - using a potentially undefined variable? We can tell that it's not set on all control flow paths

Reply:
Oops sorry, didn't see this before I merged. Let me look into fixing this.